### PR TITLE
[improvement](Nereids) Support to remove sort which is under table sink

### PR DESCRIPTION
**Pull request by seawinde**
## Proposed changes

The sort under table sink is useless, so support to remove sort which is under table sink

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

**Review by morrySnow — DISMISSED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/EliminateSort.java [hunk 1]
```diff
@@ -57,21 +59,24 @@ public Plan visit(Plan plan, Boolean pruneSort) {
     @Override
     public Plan visitLogicalSort(LogicalSort<? extends Plan> sort, Boolean eliminateSort) {
         if (eliminateSort) {
-            return visit(sort.child(), true);
+            return visit(sort.child(), eliminateSort);
         }
-        return visit(sort, true);
+        return visit(sort, eliminateSort);
     }
```
Comment (XieJiann) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/EliminateSort.java:65]: why pass `eliminateSort` rather than true

**Review by keanji-x — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/EliminateSort.java (hunk 1 above)
Comment (seawinde) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/EliminateSort.java]: have fixed it

**Comment by seawinde** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Pull request merged**